src/sqlfe/general/Submission.java

```diff
@@ -176,60 +174,114 @@ public String toString() {
-		Matcher matcher = null;						// matcher for regexp pattern matching		
 		
 		try {
 			fr = new FileReader(submissionFileName);
 			br = new BufferedReader(fr);
 			String line;
 			this.submissionFileName = submissionFileName;
 			///System.out.println("\nReading in file " + submissionFileName);
-			// TODO - remove -- -- from these three lines before writing out
-			// TODO - make how many lines are in an assignment customizable? e.g. add student id
-			line = br.readLine(); 											// get first line
-			line = Utilities.skipBlankLines(br, line);						// skip blanks if any
+			
 			final int BASE_PROMPT_LENGTH = 6;								// length of instructor comment marker >-- -- <
-			submissionName = line.substring(BASE_PROMPT_LENGTH);			// first line = assignment name, strip off leading >-- -- <
-			//System.out.println("submission name: " + submissionName);
-			line = br.readLine();											// second line = (student) name
-			final int NAME_PROMPT_LENGTH = 11;								// length of >-- -- Name:<
-			if (line.length() > NAME_PROMPT_LENGTH) {
-				studentName = line.substring(NAME_PROMPT_LENGTH).trim();	// name is whatever is after prompt
-			} else {
-				studentName = "missing";
-			}
-			//System.out.println("name: " + studentName);
-			line = br.readLine(); 											// read third line
 
-			// skip any blank lines and additional instructions for assignment (before first question instructor comments)
-			line = Utilities.skipBlankLines(br, line);
-			//System.out.println("after any blanks, next line is: >" + line + "<");
+			// Parse and store the first three lines of file which include fields class name, 
+			// student name and a blank space for maintaining separation from the next section
+			line = getFileMetadata(br);
 			
-			// if not already at the first question, look for any other instructor comments and trailing blanks and skip them
-			int attemptCount = 0;							// number of line attempts so far
-			matcher = pattern.matcher(line);
-															// stop if user comments or start of new question
-	        while (!Utilities.isUserCommentSingleLine(line) && !Utilities.isUserCommentMultiLineStart(line) && !matcher.find() && 
-	        		attemptCount <= MAX_LINE_ATTEMPTS) {						
-	        	line = Utilities.skipInstructorComments(br, line);
-	        	//System.out.println("after instructor comments, next line is: >" + line + "<");
-	        	line = Utilities.skipBlankLines(br, line);			
-	        	//System.out.println("after next set of blanks, next line is: >" + line + "<");
-	        	matcher = pattern.matcher(line);
-	        	attemptCount++;
-	        	if (attemptCount > MAX_LINE_ATTEMPTS) {
-	        		throw new SQLFEParseException("\nParse Exception in file: " + submissionFileName + ", approx. line: >" + line + "<");
-	        	}
-	        }
+			// Parse through any additional lines(empty or non-empty) before reaching the first question
+	        line = reachFirstQuestion(br, line);
```

Comment:
Need to fix indent here

Reply:
That is strange, my local changes dont have the line indented out, but I would recheck before next commit.

Reply:
I've run into similar issues with my IDE not showing tab characters. Try backspacing this line all the way over and re-indenting

Reply:
I tried to do it the way you mentioned, but I missed out other unforseen indents, so I updated all indentation in github itself.

Reply:
All good man, everything looks good